### DISCUSSION TITLE
Please implement configurable claim options

### DISCUSSION THREAD

**Discussion by raffel080108** [GriefPrevention/GriefPrevention#1749] [Ideas]
Please implement it so the owner of a claim can use /settings (or a similar command) to toggle specific settings in that specific claim (or use it outside of a claim to apply those settings to all claims).
The owner can toggle general things like can TNT explode, is PVP enabled, can monsters spawn etc.
Also implement configurable trust levels. So an owner can set "ranks" for a specific claim or all their claims, so you as an exmaple: "Admin" can do anything, "Builder" can build, open chests, pickup items and use doors and "Employee" can only kill mobs and use buttons. All of this being configurable by the owner of a claim. All of these ranks can be deleted _(except for "visitor")_, edited and newly created, as the owner desires. Via the menu the owner can, as said, either set these ranks for specific claims and have ranks for all claims in general which have all of those configurable options. Every user can be given 1 specific rank in general and/or per claim, in the case of a user having both in which case a rank for a specific claim would override general permissions

These options could be toggleable in the config, so for example a claim owner cannot toggle PVP or Monster Spawning in their claim (those options either being permanently enabled or disabled)
An example structure for this _(only a few example options, I'm not suggesting this as a full list of all toggleable things)_

**Claim-Options: true _(if disabled default values below will be used for all claims)_
  default:
   PvP: true
   Mob-Spawning: true
   TNT: false
   Other-Explosions: false
   not-toggleable:
     .- PvP
     .- Mob-Spawning**
     
   **claim-ranks: true
   defaults:
   specific-claim:
    Admin: 
      Build: true
      Resize-Claim: false
      Item-Throw-Pickup: true
      Mob-Damaging: true
      Container-Opening: true
      Door-Trapdoor-Interact: true
      Button-Lever-Interact: true
      Pressure-Plate-Interact: true
     Builder:
      Build: true
      Resize-Claim: false
      Item-Throw-Pickup: true
      Mob-Damaging: false
      Container-Opening: false
      Door-Trapdoor-Interact: true
      Button-Lever-Interact: true
      Pressure-Plate-Interact: true
     Employee:
      Build: false
      Resize-Claim: false
      Item-Throw-Pickup: true
      Mob-Damaging: true
      Container-Opening: false
      Door-Trapdoor-Interact: false
      Button-Lever-Interact: true
      Pressure-Plate-Interact: true
     Visitor: _(not be deletable via the ingame menu and if removed from config this value resets with default values)_
      Build: false
      Item-Throw-Pickup: false
      Mob-Damaging: false
      Container-Opening false
      Door-Trapdoor-Interact: false
      Button-Lever-Interact: false
      Pressure-Plate-Interact: true
all-claims:
    Admin: 
      Build: true
      Resize-Claim: true
      Item-Throw-Pickup: true
      Mob-Damaging: true
      Container-Opening: true
      Door-Trapdoor-Interact: true
      Button-Lever-Interact: true
      Pressure-Plate-Interact: true
     Visitor: _(not be deletable via the ingame menu and if removed from config this value resets with default values)_
      Build: false
      Item-Throw-Pickup: false
      Mob-Damaging: false
      Container-Opening false
      Door-Trapdoor-Interact: false
      Button-Lever-Interact: false
      Pressure-Plate-Interact: true**

I would love to have a feature like this, since it's pretty much the only thing I'm missing about the plugin. If unwanted on a server this entire feature can also be disabled in the config. _(Claim-Options: false)_
 
 Please comment on this or hit me up on discord _(raffel080108#4231)_ if more information is required to (possibly) implement this feature

**Comment by bigpresh** [comment]
This is the kind of thing that lives in https://www.spigotmc.org/resources/griefprevention-flags.55773/ - have a look at that addon as it does some of the above already I believe, see its [list of flags](https://github.com/drbot7/GriefPreventionFlags/wiki/Flags)

**Reply by raffel080108** [reply]
Thanks for the suggestion, looks like something I can try out. Still I would like to have a system with settable permissions for each player, this only helps with (general) claim flag setting, which is part of my suggestion so it does help
